### PR TITLE
DocumentMetadata v1.0.0

### PR DESCRIPTION
**Pull request by diegoalexdiaz**
Initial push for the new DocumentMetadata v1.0.0 schema.

### CODE REVIEW

**Comment by ColinKerr** [comment]
The CODEOWNERS.md file should be updated to cover ownership of this new schema

File: Domains/DocumentMetadata/DocumentMetadata.ecschema.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+<!-- ==================================================================================
+|  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+|  * See LICENSE.md in the project root for license terms and full copyright notice.
+======================================================================================= -->
+<ECSchema schemaName="DocumentMetadata" alias ="dmd" version="01.00.00" description="Metadata from a Document Management System" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+    <ECSchemaReference name="CoreCustomAttributes" version="01.00.03" alias="CoreCA"/>
+    <ECSchemaReference name="BisCore" version="01.00.14" alias="bis"/>
+
+    <ECCustomAttributes>
+        <ProductionStatus xmlns="CoreCustomAttributes.01.00.00">
```
Comment (ColinKerr) [inline Domains/DocumentMetadata/DocumentMetadata.ecschema.xml:11]: It's good hygiene to have version the the xmlns tag match the schema reference version in the ECSchemaReference.

**Review by jchick-bentley — APPROVED** [review]

File: Domains/DocumentMetadata/DocumentMetadata.ecschema.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            <SupportedUse>NotForProduction</SupportedUse>
+        </ProductionStatus>
+    </ECCustomAttributes>
+
+    <ECEntityClass typeName="DocumentLink" modifier="None" displayLabel="Document link" description="Link to a named persistent set of information, retrievable from within a file, system or application storage hierarchy.">
+        <BaseClass >bis:RepositoryLink</BaseClass>
```
Comment (abeesh) [inline Domains/DocumentMetadata/DocumentMetadata.ecschema.xml]: I am working on the connector code and I am starting to question the merits of deriving the class versus a relationship that adds metadata to a repository element. The repository link element is used widely and is used in relationships etc. Changing the type while preserving elementId is riskier.

File: Domains/1-Common/DocumentMetadata/DocumentMetadata.ecschema.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+|  * See LICENSE.md in the project root for license terms and full copyright notice.
+======================================================================================= -->
+<ECSchema schemaName="DocumentMetadata" alias ="dmd" version="01.00.00" description="Metadata from a Document Management System" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+    <ECSchemaReference name="BisCustomAttributes" version="01.00.00" alias="bisCA"/>
+    <ECSchemaReference name="CoreCustomAttributes" version="01.00.03" alias="CoreCA"/>
+    <ECSchemaReference name="BisCore" version="01.00.14" alias="bis"/>
+
+    <ECCustomAttributes>
+        <ProductionStatus xmlns="CoreCustomAttributes.01.00.03">
+            <SupportedUse>NotForProduction</SupportedUse>
```
Comment (abeesh) [inline Domains/1-Common/DocumentMetadata/DocumentMetadata.ecschema.xml:13]: Do we want to remove this custom attribute before publishing

Comment (diegoalexdiaz) [reply Domains/1-Common/DocumentMetadata/DocumentMetadata.ecschema.xml:13]: This PR puts the schema in master as the "WIP" copy. We expect all WIP copies to claim they are not ready for production.

As part of the second PR releasing this schema, we will change that flag to "production-ready" on the released copy. The WIP copy will remain as "not ready for production".

**Review by abeesh — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**